### PR TITLE
Implement find nearby station

### PR DESCRIPTION
**Pull request by duncanspumpkin**
~~I think i'm going to rework `getTrackConnections` first~~ Not tested yet

### CODE REVIEW

File: src/OpenLoco/src/World/StationManager.cpp [hunk 1]
```diff
@@ -636,6 +636,84 @@ namespace OpenLoco::StationManager
...
+                }
+                auto* station = StationManager::get(elStation->stationId());
+                if (station->owner != companyId)
+                {
+                    continue;
+                }
+                auto distDiff = World::toWorldSpace(tilePos) - pos;
+                distDiff.x = std::abs(distDiff.x);
+                distDiff.y = std::abs(distDiff.y);
+                const auto distance = std::max(distDiff.x, distDiff.y);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/World/StationManager.cpp:670]: Chebyshev distance

Comment (ZehMatt) [reply src/OpenLoco/src/World/StationManager.cpp]: Do we really need two separate ways to do distance?

**Comment by duncanspumpkin** [comment]
Replays match but unsure how much they play with this code.

File: src/OpenLoco/src/World/StationManager.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/World/StationManager.cpp]: At least while we want to retain replay compatibility with vanilla, I imagine.

**Review by AaronVanGeffen — APPROVED** [review]
Appears to be working as expected.

**Pull request merged**